### DISCUSSION TITLE
HAL_Delay in loramac stack?

### DISCUSSION THREAD

**Discussion by 13868132273** [Lora-net/LoRaMac-node#1194] [Q&A] [answered]
Hi,
The conventional practice of HAL_Delay() function is to use SysTick. But Rtc is used in the loramac stack. What other considerations do it have?

**Comment by altishchenko** [comment, marked as answer]
SysTick is usually not running in the interrupt context, for one, because of the default interrupt priority scheme used. Also, normally SysTick should not run in the powersaving mode - otherwise there will be no power saving at all, while RTC runs independently and keeps consistent time/date counting regardless of the CPU operating mode. Consider SLEEPDEEP for example, CPU is literally stopped in this state and SysTick cannot happen, while RTC ticks away nicely.

**Comment by 13868132273** [comment]
OK, thank you.@altishchenko
